app/javascript/wikiapi.js

```diff
@@ -162,6 +164,32 @@ var wikidataItem = function(spec) {
     }
   };
 
+  my.createReferences = function (claims, data) {
+    var referenceURLProp = wikidata.getPropertyID('reference URL')
+    var referenceURL = data.references[referenceURLProp]
+
+    var currentReference = getReferenceForURLFromAPIClaims(
+      getReferencesFromAPIClaims(claims, data.property),
+      referenceURLProp,
+      referenceURL
+    )
+
+    console.log('There are ' + Object.keys(data.references).length + ' references to create....');
+
+    if (Object.keys(data.references).length > 0) {
+      var data = {
+        statement: data.statement,
+        snaks: buildReferenceSnaks(data.references),
+        baserevid: lastRevisionID,
+        summary: wikidata.summary()
+      }
+      if (currentReference) data['reference'] = currentReference.hash
```

Comment:
Sorry, I feel bad mentioning style things like this when really we should have an automated tool doing it, but could we stick to the Douglas Crockford style where we always use braces after `if` conditions, and avoid squeezing things onto one line? I genuinely think it avoids misunderstandings...